ow_faults/src/FaultInjector.cpp

```diff
@@ -93,24 +104,62 @@ void FaultInjector::antennaePanFaultCb(const std_msgs::Float64& msg){
   publishAntennaeFaults(msg, 
                         m_faults.ant_pan_encoder_failure, 
                         m_faults.ant_pan_effort_failure, 
-                        m_faultPanValue, m_fault_ant_pan_pub );
+                        m_faultPanValue, m_fault_ant_pan_remapped_pub );
 }
 
 void FaultInjector::antennaeTiltFaultCb(const std_msgs::Float64& msg){
   publishAntennaeFaults(msg, 
                         m_faults.ant_tilt_encoder_failure, 
                         m_faults.ant_tilt_effort_failure, 
-                        m_faultTiltValue, m_fault_ant_tilt_pub );
+                        m_faultTiltValue, m_fault_ant_tilt_remapped_pub );
 }
 
-void FaultInjector::setPowerTemperatureFaultValue(bool getTempBool){
-  float thermal_val;
-  if (isnan(powerTemperatureOverloadValue)) {
-    thermal_val =  50.0 + static_cast <float> (rand()) /( static_cast <float> (RAND_MAX/(90.0-50.0)));
-    powerTemperatureOverloadValue = thermal_val;
-  } else if (!getTempBool) {
-    powerTemperatureOverloadValue = NAN;
+float FaultInjector::getRandomFloatFromRange( float min_val, float max_val){
+  return min_val + (max_val - min_val) * (rand() / static_cast<float>(RAND_MAX));
+}
+
+void FaultInjector::publishPowerSystemFault(){
+  //power
+  ow_faults::PowerFaults power_faults_msg;
+  ComponentFaults hardwareFault =  ComponentFaults::Hardware;
+  //system
+  std::bitset<10> systemFaultsBitmask{};
+  ow_faults::SystemFaults system_faults_msg;
+
+  systemFaultsBitmask |= systemFaultsBitset;
+  //update if fault
+  if (m_temperatureFault || m_socFault) {
+    //system
+    systemFaultsBitmask |= isPowerSystemFault;
+    systemFaultsBitset = systemFaultsBitmask;
+    //power
+    setComponentFaultsMessage(power_faults_msg, hardwareFault);
```

Comment:
To me the **systemFaultsBitmask** seems completely necessary and the code can be reduced to 
```suggestion
  ow_faults::SystemFaults system_faults_msg;

  systemFaultsBitmask |= systemFaultsBitset;
  //update if fault
  if (m_temperatureFault || m_socFault) {
    //system
    systemFaultsBitset |= isPowerSystemFault;
    //power
    setComponentFaultsMessage(power_faults_msg, hardwareFault);
    }
    setBitsetFaultsMessage(system_faults_msg, systemFaultsBitset);
```

Can you explain why the above reduced code wouldn't work?

Reply:
i believe this would overwrite any other faults that might be occurring in the system (arm, antenna) as they also get published to the system faults msg.

Reply:
You are performing an OR assignment operation not AND.. so it shouldn't clear flags that are set. You can try it of course if that would make you more comfortable.

Reply:
why does it have   systemFaultsBitmask |= systemFaultsBitset;
 but never use the bitmask var in the loop? Seems like we would only need the systemFaultsBitset |= isPowerSystemFault line?

Reply:
Yes, the line `systemFaultsBitmask |= systemFaultsBitset;` is not required .. I had fixed it in my repeated comment on the same issue.